### PR TITLE
Update Animals Table

### PR DESCRIPTION
**Pull request by ahmidou09**
## Changes Made:
- Added a column `species` of type string to the animals table.
- Inserted data for various animals.
- Implemented transactions for updating and deleting records in the animals table.

## Transactions Summary:
- Transaction 1: Set species column to unspecified and rolled back changes.
- Transaction 2: Updated species for animals with names ending in "mon" to "digimon" and set species for animals without a species already set to "pokemon".
- Transaction 3: Deleted records in the animals table and rolled back the transaction.
- Transaction 4: Deleted animals born after Jan 1st, 2022; updated weights; and rolled back to a savepoint.

## Queries Answered:
- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

## Screenshots:

- Update data sql with new insertions
01 Update data sql with new insertions

- Inserted data for various animals
02 Inserted data for various animals

- Update data sql with new insertions
03 Update data sql with new insertions

- Update the 'species' column to 'unspecified' and then rollback
Update the 'species' column to 'unspecified' and then rollback

- Update 'species' based on conditions and commit changes
05 Update 'species' based on conditions and commit changes

-  Delete all records and rollback
06 Delete all records and rollback

- Update weights and commit changes
07 Update weights and commit changes

- How many animals are there?
08 How many animals are there

- How many animals have never tried to escape?
09   How many animals have never tried to escape

- What is the average weight of animals?
10 What is the average weight of animals

- Who escapes the most, neutered or not neutered animals?
11 Who escapes the most, neutered or not neutered animals

-  What is the minimum and maximum weight of each type of animal?
12 What is the minimum and maximum weight of each type of animal

-  What is the average number of escape attempts per animal type of those born between 1990 and 2000?
13  What is the average number of escape attempts per animal type of those born between 1990 and 2000

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -23,3 +23,60 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg).
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+-- Update the 'species' column to 'unspecified' and then rollback
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals WHERE species = 'unspecified';
+ROLLBACK;
```
Comment (Strangeal) [inline queries.sql:32]: - [ ] Great work on your query to update the `species` column in the `animal` table! 👏 However, as a best practice, it's crucial to verify the success of your changes after each transaction. To ensure that the rollback operation effectively undoes the changes, I recommend querying the `animal` table following the rollback. This step adds an extra layer of confidence in the integrity of your database.

  - By executing a simple query to retrieve the data from the `animal` table after the rollback, you can confirm whether the changes were properly rolled back and that the data remains consistent.

Happy coding! 😊🚀🍀

File: queries.sql [hunk 2]
```diff
@@ -23,3 +23,60 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals WHERE species IN ('digimon', 'pokemon');
+COMMIT;
+
+-- Delete all records and rollback
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+
```
Comment (Strangeal) [inline queries.sql:49]: - [ ] Great work on your query to delete all records from the animal table! 👏 However, as a best practice, it's crucial to verify the success of your changes after each transaction. To ensure that the rollback operation effectively undoes the changes, I recommend querying the animal table following the rollback. This step adds an extra layer of confidence in the integrity of your database.

   - By executing a simple query to retrieve the data from the animal table after the rollback, you can confirm whether the changes were properly rolled back and that the data remains consistent.
   
Happy coding! 😊🚀🍀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @ahmidou09, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Well-structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @ahmidou09 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done working on this milestone so far. It is complete! 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**